optimade/server/entry_collections/elasticsearch.py

```diff
@@ -184,15 +188,28 @@ def _run_db_query(
 
         search = search.sort(*elastic_sort)
 
-        search = search[page_offset : page_offset + limit]
+        if page_offset:
+            search = search[page_offset : page_offset + limit]
+
+        elif page_above:
+            search = search.extra(search_after=page_above, limit=limit)
+
+        else:
+            search = search[0:limit]
+            page_offset = 0
+
         search = search.extra(track_total_hits=True)
         response = search.execute()
 
         results = [hit.to_dict() for hit in response.hits]
 
         if not single_entry:
             data_returned = response.hits.total.value
-            more_data_available = page_offset + limit < data_returned
+            if page_offset is not None:
+                more_data_available = page_offset + limit < data_returned
+            else:
+                # Needs to be set by some custom elastic query: is the ID the last ID?
+                more_data_available = False
```

Comment:
More data is only available, if `data_returned == len(results)`. There is the inevitable chance that you will do a request with empty response.

Reply:
I do not understand your comment.

I think `data_returned` is the total number of entries that match the filter, while `len(results)` is the number of returned entries. So if `data_returned == len(results)` all the matching entries have been returned and no more data is available.

Reply:
Just to clarify here, I was working under a potentially wonky assumption that `search_after` would mangle the total hits for a filter, but perhaps this is not the case, in which case we can still directly use `data_returned` to set `more_data_available`.

Reply:
In the documentation of Elasticsearch, they recommand setting `"track_total_hits": false` to speed things up. 
So I would expect that `data_returned` would give a sane result as we have set `"track_total_hits": true`. 
We could perhaps include the value of `data_returned` in the next link, so we can set `"track_total_hits": false` when retrieving the rest of the data.